src/__tests__/json-schema.test.js

```diff
@@ -0,0 +1,266 @@
+    jest.resetModules()
+    jest.clearAllMocks()
+  })
+
+  test.each(SEQUELIZE_JSON_TABLE)(
+    'should handle type %s',
+    (type, _sequelize, json) => {
+      const { jsonSchema } = require('../json-schema')
+      const model = sequelize.define(type, {
+        id: {
+          type: _sequelize,
+          primaryKey: true
+        }
+      })
+
+      expect(jsonSchema(model)).toEqual({
+        type: 'object',
+        properties: {
+          id: json,
+          createdAt: {
+            type: 'date'
+          },
+          updatedAt: {
+            type: 'date'
+          }
+        },
+        required: ['createdAt', 'updatedAt']
+      })
+    }
+  )
+
+  test('should handle default values', () => {
+    const Sequelize = require('sequelize')
+    const { jsonSchema } = require('../json-schema')
+
+    const defaultUuidModel = sequelize.define('default-uuid', {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true
+      }
+    })
+
+    expect(jsonSchema(defaultUuidModel)).toEqual({
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uuid',
+          default: '0ef9f424-c6f0-4bfe-b55e-87c50420180f'
+        },
+        createdAt: {
+          type: 'date',
+          default: undefined
+        },
+        updatedAt: {
+          type: 'date',
+          default: undefined
+        }
+      },
+      required: ['createdAt', 'updatedAt']
+    })
+
+    const defaultStringModel = sequelize.define('default-string', {
+      id: {
+        type: Sequelize.STRING,
+        defaultValue: 'wonkey watermelon',
+        primaryKey: true
+      }
+    })
+
+    expect(jsonSchema(defaultStringModel)).toEqual({
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          default: 'wonkey watermelon'
+        },
+        createdAt: {
+          type: 'date'
+        },
+        updatedAt: {
+          type: 'date'
+        }
+      },
+      required: ['createdAt', 'updatedAt']
+    })
+  })
+
+  test('should handle required values', () => {
+    const Sequelize = require('sequelize')
+    const { jsonSchema } = require('../json-schema')
+
+    const requiredModel = sequelize.define('required-uuid', {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true,
+        allowNull: false
+      }
+    })
+
+    expect(jsonSchema(requiredModel)).toEqual({
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uuid',
+          default: '0ef9f424-c6f0-4bfe-b55e-87c50420180f'
+        },
+        createdAt: {
+          type: 'date',
+          default: undefined
+        },
+        updatedAt: {
+          type: 'date',
+          default: undefined
+        }
+      },
+      required: ['id', 'createdAt', 'updatedAt']
+    })
+
+    const notRequiredModel = sequelize.define('not-required', {
```

Comment:
Why this case? What's different about it?

Reply:
removed